util/types.c

```diff
@@ -78,6 +79,18 @@ char *uint128_t_to_string(nvme_uint128_t val)
 	return str + idx;
 }
 
+char *int128_to_si_string(__u8 *data, __u32 bytes_per_unit)
+{
+	static char str[40];
+	long double val = int128_to_double(data) * bytes_per_unit;
```

Comment:
Does this work on 32bit machines? 

BTW, I would like to ask you to add unit tests for these functions, like the ones in `tests/test-uint128.c`.

Reply:
Yes I will do test on 32bit environment and add unit tests as mentioned.
Note: My local 32bit enbironment is virtual environment as I used 32bit Ubuntu as guest OS on virtual box but use 64bit PC and Windows host OS.

Reply:
Tested on 32 bit environment and confirmed as works correctly.